test/integration/master/kms_transformation_test.go

```diff
@@ -140,22 +122,79 @@ func TestKMSProvider(t *testing.T) {
+// Cacher.go (assuming sufficient memory) should intercept requests for secrets that were recently created or recently retrieved - thus
+// alleviating the need for caching within envelop.go.
+func TestKMSFailureImpactOnReads(t *testing.T) {
+	pluginMock, test := mustSetupTest(t)
+	defer pluginMock.cleanUp()
+	defer test.cleanUp()
+
+	pluginMock.gracefulStop()
+	// Expect hit - cache was populated when the test secret was created (i.e. cacher.go has a cached version of the secret in plain text).
+	mustReadSecret(t, test, false)
+}
+
+func mustSetupTest(t *testing.T) (*base64Plugin, *transformTest) {
+	pluginMock, err := NewBase64Plugin()
+	if err != nil {
+		t.Fatalf("failed to create mock of KMS Plugin: %v", err)
+	}
+
+	test, err := newTransformTest(t, kmsConfigYAML)
+	if err != nil {
+		pluginMock.cleanUp()
+		t.Fatalf("failed to start KUBE API Server with encryptionConfig\n %s", kmsConfigYAML)
+	}
+
+	// As part of newTransformTest a new secret was created, so KMS Mock should have been exercised by this point.
+	if len(pluginMock.errorChan) != 0 {
+		pluginMock.cleanUp()
+		test.cleanUp()
+		t.Fatalf("KMSPlugin failed while serving requests: %v", <-pluginMock.errorChan)
+	}
+
+	// Ensuring that the test secret has been cached - written to etcd and read back by cacher.
+	pollErr := wait.PollImmediate(1*time.Second, wait.ForeverTestTimeout, func() (bool, error) {
+		_, err := test.restClient.CoreV1().Secrets(test.ns.Name).Get(test.secret.Name, metav1.GetOptions{ResourceVersion: "0"})
+		return err==nil, nil
+	})
+
+	if pollErr == wait.ErrWaitTimeout {
+		pluginMock.cleanUp()
+		t.Fatalf("failed to retrieve secret from cache within the alloted time period: %d", wait.ForeverTestTimeout)
+	}
+
+	return pluginMock, test
+}
+
+func mustReadSecret(t *testing.T, test *transformTest, bypassCache bool) {
+	// ResourceVersion 0 implies cache will be checked, "" implies that cache will be bypassed.
+	resourceVersion := "0"
+	if bypassCache {
+		resourceVersion = ""
+	}
+
+	s, err := test.restClient.CoreV1().Secrets(test.ns.Name).Get(test.secret.Name, metav1.GetOptions{ResourceVersion: resourceVersion})
+	if err != nil {
+		t.Fatalf("failed to get %s, error: %v", test.secret.Name, err)
+	}
 	if secretVal != string(s.Data[secretKey]) {
 		t.Fatalf("expected %s from KubeAPI, but got %s", secretVal, string(s.Data[secretKey]))
 	}
-	test.printMetrics()
 }
 
 func getDEKFromKMSPlugin(pluginMock *base64Plugin) ([]byte, error) {
-	// We expect KMS to already have seen an encryptRequest. Hence non-blocking call.
-	e, ok := <-pluginMock.encryptRequest
-
-	if !ok {
-		return nil, fmt.Errorf("failed to sense encryptRequest from KMS Plugin Mock")
+	select {
+	case e := <-pluginMock.encryptRequest:
+		return e.Plain, nil
+	case <-time.After(wait.ForeverTestTimeout):
+		return nil, fmt.Errorf("expected encryptRequest to be received by KMS Plugin within %d seconds", wait.ForeverTestTimeout)
```

Comment:
replace "%d seconds" with "%v", wait.ForeverTestTimeout should be time.Duration

Reply:
Done.